test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/Batch/AsyncMethodTests.cs

```diff
@@ -0,0 +1,132 @@
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.OData;
+using Microsoft.AspNetCore.OData.Batch;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.Tests.Batch.Server;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Microsoft.OData.Edm.Csdl;
+using System.Xml;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.Tests.Batch
+{
+    public class AsyncMethodTests : EndToEndTestBase<AsyncMethodTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(BanksController), typeof(MetadataController));
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel(), new DefaultODataBatchHandler()));
+            }
+        }
+
+        public AsyncMethodTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri)
+            {
+                HttpClientFactory = HttpClientFactory
+            };
+        }
+
+        [Fact]
+        public async Task JsonBatchSequencingSingeChangeSetTest()
+        {
+            // Create new BankAccounts object
+            var bank = new Bank
+            {
+                Id = 45,
+                Name = "Test Bank",
+                Location = "KE",
+                BankAccounts = new List<BankAccount>()
+            };
+
+            // Create new BankAccount object
+            var bankAccount = new BankAccount
+            {
+                Id = 890,
+                AccountNumber = "4567890",
+                BankId = bank.Id,
+                Bank = bank
+            };
+
+            // Establish the relationship between BankAccounts and BankAccount
+            bank.BankAccounts.Add(bankAccount);
+
+            // Add the BankAccounts entity to the context
+            _context.AddObject("Banks", bank);
+
+            // Add the related BankAccount entity
+            _context.AddRelatedObject(bank, "BankAccounts", bankAccount);
+
+            // Save both entities in a single batch request using JSON
+            var response = await _context.SaveChangesAsync(SaveChangesOptions.BatchWithSingleChangeset | SaveChangesOptions.UseJsonBatch);
+            Assert.Equal(201, (response.First() as ChangeOperationResponse).StatusCode);
+            Assert.Equal(201, (response.Last() as ChangeOperationResponse).StatusCode);

```

Comment:
```suggestion
            Assert.Equal(201, Assert.IsType<ChangeOperationResponse>(response.First()).StatusCode);
            Assert.Equal(201, Assert.IsType<ChangeOperationResponse>(response.Last()).StatusCode);
```
- safer - avoid `NullReferenceException` in your test
- neater

Reply:
I addition, I suggest you add an assert statement above the two statements to verify that `result` has at least two elements, otherwise `.First()` can also be the `.Last()`. We want to confirm that both POST requests for Bank and BankAccount happened